app/src/ui/diff/modified-image-diff.tsx

```diff
@@ -1,25 +1,342 @@
 import * as React from 'react'
 
+import { ImageDiffType } from '../../lib/app-state'
 import { Image } from '../../models/diff'
 import { renderImage } from './render-image'
+import { TabBar, TabBarType } from '../tab-bar'
 
 interface IModifiedImageDiffProps {
   readonly previous: Image
   readonly current: Image
+  readonly diffType: ImageDiffType
+  readonly onChangeDiffType: (type: number) => void
+}
+
+const SIZE_CONTROLS = 60
+const PADDING = 20
+
+const getDimensions = (
+  naturalHeight: number | null,
+  naturalWidth: number | null,
+  _containerWidth: number,
+  _containerHeight: number
+) => {
+  // keep some padding
+  const containerWidth = _containerWidth - PADDING
+  const containerHeight = _containerHeight - PADDING - SIZE_CONTROLS
+
+  // check wether we will need to scale the images or not
+  const heightRatio =
+    containerHeight < (naturalHeight || 0)
+      ? (naturalHeight || 0) / containerHeight
+      : 1
+  const widthRatio =
+    containerWidth < (naturalWidth || 0)
+      ? (naturalWidth || 0) / containerWidth
+      : 1
+
+  // Use max to prevent scaling up the image
+  let ratio = Math.max(1, widthRatio)
+  if (widthRatio < heightRatio) {
+    // fit to height
+    ratio = Math.max(1, heightRatio)
+  }
+
+  return {
+    width: (naturalWidth || 0) / ratio,
+    height: (naturalHeight || 0) / ratio,
+  }
 }
 
 /** A component which renders the changes to an image in the repository */
 export class ModifiedImageDiff extends React.Component<
   IModifiedImageDiffProps,
-  {}
+  {
+    value: number
+    naturalWidthBefore: number | null
+    naturalHeightBefore: number | null
+    naturalWidthAfter: number | null
+    naturalHeightAfter: number | null
+  }
 > {
+  private _container: HTMLDivElement | null
+
+  public constructor(props: IModifiedImageDiffProps) {
+    super(props)
+    this.state = {
+      value: 1,
+      naturalWidthBefore: null,
+      naturalHeightBefore: null,
+      naturalWidthAfter: null,
+      naturalHeightAfter: null,
+    }
+  }
+
+  private handleValueChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      value: parseFloat(e.currentTarget.value),
```

Comment:
Why `parseFloat` instead of `parseInt`? I'm not totally sure what's going on here.